cmd/provisioner/deletelv.go

```diff
@@ -60,23 +62,40 @@ func deleteLV(c *cli.Context) error {
 
 	klog.Infof("delete lv %s vg:%s dir:%s block:%t", lvName, vgName, dirName, blockMode)
 
-	output, err := umountLV(lvName, vgName, dirName)
+	found := false
+	lvs, err := commands.ListLV(context.Background(), vgName)
 	if err != nil {
-		return fmt.Errorf("unable to delete lv: %w output:%s", err, output)
+		return fmt.Errorf("unable to list existing logicalvolumes:%v", err)
+	}
+	for _, lv := range lvs {
+		if strings.Contains(lv.Name, lvName) {
```

Comment:
Why contains and not `==`? Is contains always safe?

Reply:
this was because sometimes lvm converts a single dash `-` in the lv name to double dash `--` if i remember correctly

Reply:
Okay, never mind then. Just looked suspicious at first glance, also don't understand how `--` match, but I guess when it worked all the time before, it will be fine. :)